examples/README.md

```diff
@@ -0,0 +1,544 @@
+    memcpy(response, result.c_str(), neededSize);
+    *actual_response_len = neededSize;
+    LOG_DEBUG("HelloworldCC: Response: %s", result.c_str());
+    LOG_DEBUG("HelloworldCC: +++ Executing done +++");
+    return 0;
+}
+```
+
+#### Build 
+
+Make sure you have the [environment variables](https://github.com/hyperledger-labs/fabric-private-chaincode#environment-settings) set.  In addition, set `SGX_BUILD=DEBUG` to enable log messages.
+
+To build the helloworld chaincode, we are using CMake. This simplifies the build process and compiles our chaincode using the SGX SDK. Create `CMakeLists.txt` with the following content.
+
+File: CMakeLists.txt
+```
+cmake_minimum_required(VERSION 3.5.1)
+
+set(SOURCE_FILES
+    helloworld_cc.cpp
+    )
+
+include(../../ecc_enclave/enclave/CMakeLists-common-app-enclave.txt)
+```
+
+Create `Makefile` with the following content.  For your convenience, you can copy the `Makefile` from FPC-INSTALL-DIR/examples/auction folder.
+File: examples/helloworld/Makefile 
+```
+TOP = ../..
+include $(TOP)/build.mk
+
+BUILD_DIR := _build
+
+$(BUILD_DIR):
+	@if [ ! -d $(BUILD_DIR) ]; then \
+		mkdir -p $(BUILD_DIR) && \
+		cd $(BUILD_DIR) && \
+		cmake ./..; \
+	fi
+
+build: $(BUILD_DIR)
+	$(MAKE) --directory=$<
+
+clean:
+	-rm -rf $(BUILD_DIR)
+```
+
+
+In FPC-INSTALL-DIR/examples/helloworld folder, to build the chaincode, execute: 
+```
+make
+```
+
+Following is a part of expected output.  Please note `[100%] Built target enclave` message in the output.  This suggests that build was successful. 
+Output:  
+```
+make[3]: Leaving directory '/home/bcuser/work/src/github.com/hyperledger-labs/fabric-private-chaincode/examples/helloworld/_build'
+[100%] Built target enclave
+make[2]: Leaving directory '/home/bcuser/work/src/github.com/hyperledger-labs/fabric-private-chaincode/examples/helloworld/_build'
+/usr/bin/cmake -E cmake_progress_start /home/bcuser/work/src/github.com/hyperledger-labs/fabric-private-chaincode/examples/helloworld/_build/CMakeFiles 0
+make[1]: Leaving directory '/home/bcuser/work/src/github.com/hyperledger-labs/fabric-private-chaincode/examples/helloworld/_build'
+```
+
+
+#### Time to test !
+  Next step is to test the chaincode by invoking the transactions, for which you need a basic Fabric network with a channel and Enclave Registry setup. You will use the FPC test framework to bring up a Fabric network in which the helloworld code can be executed as a chaincode _in an SGX enclave_.  
+  
+  Create a file `test.sh` in examples/helloworld folder as follows.  Note that the initial lines in the script points to files and folders in FPC framework.  
+-`FPC_TOP_DIR` points to FPC-INSTALL-DIR
+-`CONFIG_HOME` points to the FPC-INSTALL-DIR/integration/config, which contains yaml files that define the Fabric network
+-`FABRIC_SCRIPTDIR` points to scripts with custom FPC wrappers and utility scripts.  For example, ledger_init and and ledger_shutdown are used to bring up and shut down the network
```

Comment:
this is a bit misleading: ledger_init/ledger_shutdown is a function imported from common_ledger.sh, the commands/scripts to do it interactively are called ledger_init.sh and ledger_shutdown.sh.    But maybe best having a separate section on interactive use of FPC where we mention the scripts and also emphasize that the peer.sh from there _must_ be used as use of "standard" fabric peer binary will fail.

Some related notes:
- in the change of PR144, ledger_init.sh also prints commands to set the environment variables like CHAN_ID, PEER_CMD and alike which come handy when running interactively tests
- the commands now also honor a already pre-existing environment variable FABRIC_CFG_PATH (previously, you had to run peer.sh inside the config directory ...), ledger_init.sh should also honor CHAN_ID if defined (e.g., if you want a different one to the default mychannel).